src/runtime/list.cpp

```diff
@@ -439,7 +439,20 @@ extern "C" Box* listSetitemSlice(BoxedList* self, BoxedSlice* slice, Box* v) {
         v_size = tv->size();
         v_elts = &tv->elts[0];
     } else {
-        RELEASE_ASSERT(0, "unsupported type for list slice assignment: '%s'", getTypeName(v));
+        BoxedList* l = new BoxedList();
+        try {
+            for (auto* b : v->pyElements())
+                listAppendInternal(l, b);
+        } catch (ExcInfo e) {
+            raiseExcHelper(TypeError, "can only assign an iterable");
```

Comment:
Hmm I'm assuming that this is just to catch errors from pyElements(), but will this also swallow any exceptions ex that are thrown inside a generator?